### PR TITLE
Replace dicts with maps

### PR DESCRIPTION
**Pull request by elbrujohalcon**
but leave it alone where Dialyzer itself returns a `dict`.

ℹ️  I created this draft PR for you, @maco … so we can talk about it inline.

### CODE REVIEW

File: src/rebar3_mini_typer.erl [hunk 1]
```diff
@@ -272,7 +272,7 @@ write_inc_files(Inc, Analysis) ->
              msg(debug, "Records ~tp", [Info#info.records], Analysis),
              write_typed_file(File, Info, Analysis)
           end,
-    lists:foreach(Fun, dict:fetch_keys(Inc#inc.map)).
```
Comment (elbrujohalcon) [inline src/rebar3_mini_typer.erl:275]: My idea was to actually remove the middle-man/abstraction.
Stop using `map_dict*` functions and directly use maps wherever possible.
As you can see on this line, for instance, the abstraction was broken anyway (They used a `dict` function on something that they shouldn't had known to be a map).

File: src/typer_core.erl [hunk 2]
```diff
@@ -51,10 +51,10 @@
...
          %% as {FileName, ModuleName} in case the ModuleName is different
          fms = [] :: [{file:filename(), module()}],
-         ex_func = map_dict_new() :: map_dict(),
-         record = map_dict_new() :: map_dict(),
-         func = map_dict_new() :: map_dict(),
-         inc_func = map_dict_new() :: map_dict(),
+         ex_func = maps:new() :: map(),
+         record = maps:new() :: map(),
+         func = maps:new() :: map(),
+         inc_func = maps:new() :: map(),
```
Comment (elbrujohalcon) [inline src/typer_core.erl:57]: Can we be more precise here, using stuff like `#{atom() => …}` instead of just the generic `map()`?

**Review by elbrujohalcon — COMMENTED** [review]
I think that the only thing missing here is an improvement on the type specification for the record fields.

**Comment by elbrujohalcon** [comment]
@maco Since I opened this PR… I can only comment on it (i.e. I can't approve)… but, other than the specificity in the record files (that we can leave for later), I think it's mergeable.

**Comment by pablocostass** [comment]
@elbrujohalcon I would oppose to merging the PR until we get to benchmark the (possible? expected?) improvements of replacing dicts with maps. Better to end up fine-tuning, if needed, this PR than having to clutter up the PR section with hits of changes for this.

Besides, getting some numbers would be something fancy for a blog post :P

**Review by pablocostass — APPROVED** [review]

**Pull request merged**